command/agent/agent_endpoint.go

```diff
@@ -452,6 +452,14 @@ func (s *HTTPServer) AgentRegisterService(resp http.ResponseWriter, req *http.Re
 		return nil, nil
 	}
 
+	// Check the service address here and in the catalog RPC endpoint
+	// since service registration isn't sychronous.
+	if args.Address == "0.0.0.0" {
```

Comment:
We need this in two packages `command/agent` and `consul`. I think we're good on this one.